xrayvision/visibility.py

```diff
@@ -406,6 +438,53 @@ def instrument(self):
             i += 1
         return instr
 
+    @property
+    def __eq__(self, other):
+        if not _attrs_both_none_or_neither(self.observer_coordinate, other.observer_coordinate):
+            return False
+        if hasattr(self.observer_coordinate, "__len__"):
+            if not hasattr(other.observer_coordinate, "__len__"):
+                if all(self.observer_coordinate == other.observer_coordinate):
+                    return False
+            else:
+                return False
+        elif hasattr(other.observer_coordinate, "__len__"):
+            return False
+        elif self.observer_coordinate != other.observer_coordinate:
+            return False
+
+        if not _attrs_both_none_or_neither(self.vis_labels, other.vis_labels) or tuple(self.vis_labels) != tuple(
+            other.vis_labels
+        ):
+            return False
+
+        if not _attrs_both_none_or_neither(self.instrument, other.instrument) or tuple(self.instrument) != tuple(
+            other.instrument
+        ):
+            return False
+
+        if not (
+            _attrs_both_none_or_neither(self.spectral_range, other.spectral_range)
+            and apu.quantity.allclose(self.spectral_range, other.spectral_range)
+        ):
+            return False
+
+        if not (
+            _attrs_both_none_or_neither(self.time_range, other.time_range)
+            and np.allclose(self.time_range.mjd == other.time_range.mjd)
+        ):
+            return False
+        return True
```

Comment:
There are a couple justifications for not checking those:
1. Two metas whose `VisMetaABC` attributes are equal, are equal `VisMetaABC` objects.  Values outside the definition of the `ABC` are out of scope.
2. Since these values aren't defined, and given the flexibility of types that `dict` can take, there isn't an obvious generalised way to check that the values are equal.  Some may require `val1 == val2`, others may require `all(val1 == val2)` or `assert_quantity_allclose(val1, val2)` or `assert np.allclose(val1, val2)`, or perhaps something more custom for custom objects.  Given this, it's well-defined to stick to just the `VisMetaABC`-defined attributes.

Reply:
As long as we clearly document sounds good.